coalib/bears/Bear.py

```diff
@@ -288,6 +288,14 @@ def execute(self, *args, debug=False, **kwargs):
         name = self.name
         try:
             self.debug('Running bear {}...'.format(name))
+
+            # If `dependency_results` kwargs is defined but there's no
+            # dependency results (usually in Bear that has no dependency)
+            # delete the `dependency_results` kwargs, since most Bears doesn't
+            # define `dependency_results` kwargs in its `run()` argument.
+            if kwargs.get('dependency_results', False) is None:
```

Comment:
Hmm this optional value fallback is quite a tricky piece of code ;)

What about rephrasing to
```python
if 'dependency_results' in kwargs and kwargs['dependency_results'] is None:
    del ...
```
That's a bit more obvious to grasp (imo)

Up to you :+1: